testing/cli/test_cli_sct_analyze_lesion.py

```diff
@@ -1,16 +1,71 @@
+    """Define a fake voxel lesion using the specified dimensions."""
+    # TODO: Parametrize this fixture to test other lesion dimensions
+    starting_coord = [29, 27, 25]
+    dim = [1, 4, 1]
+
+    # Format the coordinates into a str argument that `-create` can accept
+    coordinates = []
+    for x in range(starting_coord[0], starting_coord[0] + dim[0]):
+        for y in range(starting_coord[1], starting_coord[1] + dim[1]):
+            for z in range(starting_coord[2], starting_coord[2] + dim[2]):
+                coord = [str(x), str(y), str(z), "1"]
+                coordinates.append(",".join(coord))
+    create_arg = ":".join(coordinates)
+
+    # Create the lesion mask file and output to a temporary directory
+    path_ref = sct_test_path("t2", "t2.nii.gz")
+    path_out = str(tmp_path/"lesion.nii.gz")
+    sct_label_utils.main(argv=['-i', path_ref, '-o', path_out,
+                               '-create', create_arg])
+
+    # Compute the expected measurements from the provided dimensions
+    measurements = {
+        # NB: 'sct_analyze_lesion' treats lesions as cylinders. So:
+        #   - Y axis: Length of the cylinder
+        'length [mm]': dim[1],
+        #   - X/Z plane: Cross-sectional slices of the cylinder.
+        #                Specifically, 'max_equivalent_diameter' takes the X/Z
+        #                cross-sectional area of the lesion (which is computed
+        #                using square voxels), then finds the diameter of an
+        #                equivalent *circle* with that same area:
+        #                   a = pi*r^2
+        #                -> a = pi*(d/2)^2
+        #                -> d = 2*sqrt(a/pi)
+        'max_equivalent_diameter [mm]': 2 * np.sqrt(dim[0] * dim[2] / np.pi),
+        'volume [mm3]': dim[0] * dim[1] * dim[2],
+    }
+
+    return path_out, measurements
+
+
 @pytest.mark.sct_testing
-@pytest.mark.usefixtures("run_in_sct_testing_data_dir")
-def test_sct_analyze_lesion_output_file_exists():
+def test_sct_analyze_lesion_output_file_exists(dummy_lesion, tmp_path):
     """Run the CLI script and verify output file exists."""
-    sct_analyze_lesion.main(argv=['-m', 't2/t2_seg-manual.nii.gz', '-s', 't2/t2_seg-manual.nii.gz'])
-    assert os.path.exists('t2_seg-manual_analyzis.pkl')
+    # Run the analysis on the dummy lesion file
+    path_lesion, expected_measurements = dummy_lesion
+    sct_analyze_lesion.main(argv=['-m', path_lesion,
+                                  '-s', sct_test_path("t2", "t2_seg-manual.nii.gz"),
+                                  '-ofolder', str(tmp_path)])
+
+    # Load analysis results from pickled pandas.Dataframe
+    _, fname, _ = extract_fname(path_lesion)
+    with open(tmp_path/f"{fname}_analyzis.pkl", 'rb') as f:
+        measurements = pickle.load(f)['measures']
+
+    # Validate analysis results
+    for key, expected_value in expected_measurements.items():
+        assert(measurements.at[0, key] == expected_value)
```

Comment:
There's a slight snag here. For the `length [mm]` measurement, there's a mismatch between what gets logged in the console, and the _actual_ value.

Log:

```
Averaged measures...
  Volume [mm^3] = 4.0+/-0.0
  (S-I) Length [mm] = 4.0+/-0.0
  Equivalent Diameter [mm] = 1.13+/-0.0

Total volume = 4.0 mm^3
Lesion count = 1
```

Actual value:

![Screenshot from 2022-02-05 10-44-43](https://user-images.githubusercontent.com/16181459/152648540-a4329bd8-bb3b-4d6d-a7e7-e0a0d39d6d7c.png)

This is why the test fails. But, I'm pretty sure this is an issue with the script, rather than the test.

Reply:
Yes, I think this is because the logged values are rounded ([using `np.round` here](https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/f0b244cd174463a3825620bc488767c409beccdc/spinalcordtoolbox/scripts/sct_analyze_lesion.py#L219)).

As a separate point, it's a bit alarming that the uncertainties as `+/-0.0`. Surely they should be rounded up to `0.1`?

Based on the computations in the script, we expect:
- the volume to be exactly 4.0, aka the number of voxels in the lesion;
- the length to be a little bit more than 4.0, because it's measured along the curve of the spinal cord centerline (if I understand correctly); if so, that's another bug uncovered, and [this line](https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/f0b244cd174463a3825620bc488767c409beccdc/spinalcordtoolbox/scripts/sct_analyze_lesion.py#L251) should be _dividing_ by the cosine, rather than multiplying;
- the diameter to be a little bit less than 2/sqrt(pi) = 1.1283791670955126, again because there's a correction factor that takes into account the spinal cord centerline.

Reply:
Hrmm. I tried playing with some different lesion sizes and got consistently less than the expected length, regardless of where I placed the lesion. For reference, here is the t2 image, centerline, and a sample lesion I tried:

![Screenshot from 2022-02-08 13-18-55](https://user-images.githubusercontent.com/16181459/153050900-6a46de32-a71d-43be-a68f-dfbea0d3d9d0.png)

Reply:
Resolved by https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3681#issuecomment-1035562858.